pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/TransformerWorker.scala

```diff
@@ -100,10 +108,14 @@ trait TransformerWorker[In, SenderDest] extends Logging {
                 }
                 Future.failed(err)
               }
-              case Right((work, key)) => {
+              case Success(key, work) => {
                 info(s"$name: from $key transformed $work")
                 Future.successful(message)
               }
+              case Surpress(key) => {
```

Comment:
I like this - but it break the rule @jamieparkinson and I have spoken about of 1 in 1 out - where there is a source version, we should always go right through to ingestion.

We have "suppressed" works previously by using `InvisibleWorks` - should we stick to this way of doing things?

Reply:
Ah didn't know that rule was something we wanted.

The reason I didn't try and transform to invisbile work is because the way it appeared when discussing with @jtweed is that the work was not parsable and hence the transformer would fail. He said works with `Tranmission=No` don't even make it to Sierra, so if that is the case shouldn't be coming through the pipeline on that side. Any thoughts @jtweed?

Reply:
My 2¢ - I don't really mind how we do it, but we have it one way, and would prefer to have only 1 way.

[Here's an example test with Sierra](https://github.com/wellcomecollection/catalogue/blob/2c57aec3571326ba34adbe1d61c10e4db4c67133/pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/SierraTransformableTransformerTest.scala#L291).

Reply:
I am kind of OK with suppressing things at the transformer level as at this stage I would argue that they are not works - they are source data.

However - it's spelled `Suppress` 😛

Reply:
The other consideration is if someone goes from a Transmittable record, to setting it to non-transmittable, we should have that pass through the pipeline, and render the appropriate message on the site. i.e. this item is now suppressed.

Reply:
I can see why we would want to do that, but don't think that is possible with the current pipeline architecture? We would have to have the record history as input to the transformer to indicate something has gone from non-suppresses to suppressed.

Reply:
That's why we just transform it to an invisible work if it's suppressed, so it would override the current one, allowing us to respond correctly from the API.

e.g. A user in calm does these steps.
```
1. CalmRecord.Transmission = "No"  => InvisibleWork
2. CalmRecord.Transmission = "Yes" => TransformedWork
3. CalmRecord.Transmission = "No" => InvisibleWork
```

Reply:
Ah I get you, yes would say the work is surpressed, thought you meant would display it's history somehow.

Reply:
Na, just that it is _no longer_ available.

Reply:
Well my point was that the user won't know whether an invisible work is no longer available or never available, but in practices thats probably fine

Reply:
Yea, a way to get around having to know previous state is always send suppressed as Invisible - agreed, the type of unavailable is not currently possible or desired.

We currently just say something has been deleted:
https://api.wellcomecollection.org/catalogue/v2/works/qefk954w